src/utils/deploy/templates.ts

```diff
@@ -28,9 +28,21 @@ const deployTemplates = async (artifactDirectory: string, config: Configstore) =
     return;
   }
 
-  await deployTemplateTypes(artifactDirectory, TemplateType.Decision, config);
-  await deployTemplateTypes(artifactDirectory, TemplateType.Web, config);
-  await deployTemplateTypes(artifactDirectory, TemplateType.Audience, config);
+  switch(templateType) {
+    case TemplateType.Audience:
+      await deployTemplateTypes(artifactDirectory, TemplateType.Audience, config);
+      break;
+    case TemplateType.Decision:
+      await deployTemplateTypes(artifactDirectory, TemplateType.Decision, config);
+      break;
+    case TemplateType.Web:
+      await deployTemplateTypes(artifactDirectory, TemplateType.Web, config);
+      break;
+    default: 
+      await deployTemplateTypes(artifactDirectory, TemplateType.Audience, config);
+      await deployTemplateTypes(artifactDirectory, TemplateType.Decision, config);
+      await deployTemplateTypes(artifactDirectory, TemplateType.Web, config);
+  }
```

Comment:
I feel like this switch statement might be breaking a pattern.  I wonder if there is a better way to handle.  Like maybe if all then run all else, just pass the template type to run.  That way if a new template type is added in the future (which I would say is a pretty high likelihood), then adding to the all scenario is all that's needed. In fact you might be able to somehow loop through the combinations and right this so you wouldn't have to write any code.  It is good though to call out, that offer templates will be handled differently here.

Reply:
1. Refactored to loop through TemplateType enum for TemplateType.All
2. Else, check if param passed in exists in the enum before calling deployTemplateTypes
3. Add Offer template type - added note to implement offer templates
4. Refactored logline(chalk.red(...)) to `logError`